osu.Game/Beatmaps/Formats/OsuLegacyDecoder.cs

```diff
@@ -220,7 +222,7 @@ private void handleDifficulty(Beatmap beatmap, string line)
         /// <param name="line">The line which may contains variables.</param>
         private void decodeVariables(ref string line)
         {
-            while (line.IndexOf('$') >= 0)
+            if (line.IndexOf('$') >= 0)

```

Comment:
I tried to fix this on a previous pr, however the uglyness got me and cound do anything
[This is the file I was testing with](https://github.com/ppy/osu/files/1161478/Deorro.-.Yee.zip)